OneCommunityGlobal/HighestGoodNetworkApp
# Koushica bell notification

# Description
This pull request resolves the bell notification issue for hours due in 48 hours and includes several new features and improvements.
<img width="701" alt="Screenshot 2024-12-17 at 5 47 58 PM" src="https://github.com/user-attachments/assets/47a5c5e3-793b-45e1-be45-db7b6d692f6e" />

This PR is an update to https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2805, here the code is little optimized and solved merge conflicts.

## Related PRS (if any):
This frontend PR is related to the development backend PR.

## How to test:
1. Check into the current branch.
2. Run npm install.
3. Clear Site Data/Cache: This will ensure no old notifications interfere.
4. Log in as any user.
5. Navigate to the Dashboard and verify the notification bell functionality.
6. Use the "Add Intangible Hours" function in the Timelog to manipulate the logged time.
7. Verify that the bell notification triggers when less than 50% of hours are logged and only 48 hours remain.
8. Confirm the notification resets after it is viewed or when the week ends.
9. Ensure the notification popup works properly in both light and dark modes.

## Note:
1. To test different time thresholds for notifications, modify the hoursLeft value in the BellNotification.jsx component.
2. Use the "Add Intangible Hours" feature to simulate different time log scenarios.
3. The notification triggers only once per week. Clear cookies or localStorage to reset the notification for retesting.
